pages/desktop/base.py

```diff
@@ -183,6 +202,156 @@ def click_logout(self):
         action.perform()
         self.wait.until(lambda s: self.is_element_displayed(*self._login_locator))
 
+    def click_user_menu_collections_link(self):
+        user = WebDriverWait(
+            self.selenium, 30, ignored_exceptions=StaleElementReferenceException
+        ).until(
+            EC.element_to_be_clickable(
+                (
+                    By.CSS_SELECTOR,
+                    '.Header-user-and-external-links \
+            .DropdownMenu-button-text',
+                )
+            )
+        )
+        link = self.find_element(*self._user_menu_collection_link_locator)
+        dropdown = self.find_element(*self._account_dropdown_locator)
+        loop = 0
+        while loop < 3:
+            try:
+                action = ActionChains(self.selenium)
+                action.move_to_element(user)
+                action.pause(2)
+                action.move_to_element(dropdown)
+                action.move_to_element(link)
+                action.pause(2)
+                action.click()
+                action.perform()
+                # waits for the user collections page to open
+                self.wait.until(
+                    EC.visibility_of_element_located(
+                        (By.CSS_SELECTOR, '.CollectionList-info')
+                    )
+                )
+                break
+            except (StaleElementReferenceException, WebDriverException) as error:
+                print(f'{error} Retrying action chains')
+            loop += 1
+
+    def click_user_menu_view_profile(self):
+        user = WebDriverWait(
+            self.selenium, 30, ignored_exceptions=StaleElementReferenceException
+        ).until(
+            EC.element_to_be_clickable(
+                (
+                    By.CSS_SELECTOR,
+                    '.Header-user-and-external-links \
+            .DropdownMenu-button-text',
+                )
+            )
+        )
+        link = self.find_element(*self._user_menu_view_profile_locator)
+        dropdown = self.find_element(*self._account_dropdown_locator)
+        loop = 0
+        while loop < 1:
+            try:
+                action = ActionChains(self.selenium)
+                action.move_to_element(user)
+                action.pause(2)
+                action.move_to_element(dropdown)
+                action.move_to_element(link)
+                action.pause(2)
+                action.click()
+                action.perform()
+                # waits for the View profile page to open
+                self.wait.until(
+                    EC.visibility_of_element_located(
+                        (By.CSS_SELECTOR, '.UserProfile-name')
+                    )
+                )
+                break
+            except (StaleElementReferenceException, WebDriverException) as error:
+                print(f'{error} Retrying action chains')
+            loop += 1
```

Comment:
These actions seem repeated between tests with some changes to the locators. We could make a fixture factory and just feed the fixture the locators that change between tests.  See: https://docs.pytest.org/en/6.2.x/fixture.html#factories-as-fixtures

Reply:
What do you think? Looks like there are some differences but we can pass the needed changes to the function.